src/reports/library/hdf5_writer.cpp

```diff
@@ -0,0 +1,204 @@
+#include <iostream>
+
+#include "reportinglib.hpp"
+#include "hdf5_writer.hpp"
+
+HDF5Writer::HDF5Writer(const std::string& report_name)
+: IoWriter(report_name), m_file(0), m_dataSet(0) {
+
+    hid_t plist_id = H5Pcreate(H5P_FILE_ACCESS);
+
+#ifdef HAVE_MPI
+    // Enable MPI access
+    MPI_Info info = MPI_INFO_NULL;
+    H5Pset_fapl_mpio(plist_id, ReportingLib::m_allCells, info);
+
+    // Initialize independent/collective lists
+    m_collective_list = H5Pcreate(H5P_DATASET_XFER);
+    m_independent_list = H5Pcreate(H5P_DATASET_XFER);
+    H5Pset_dxpl_mpio(m_collective_list, H5FD_MPIO_COLLECTIVE);
+    H5Pset_dxpl_mpio(m_independent_list, H5FD_MPIO_INDEPENDENT);
+#endif
+
+    // Create hdf5 file named after the report_name
+    m_file = H5Fcreate(report_name.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, plist_id);
+    m_offset[0] = 0;
+    m_offset[1] = 0;
+
+    H5Pclose(plist_id);
+}
+
+HDF5Writer::~HDF5Writer() {}
+
+void HDF5Writer::configure_group(const std::string& group_name) {
+    hid_t group = H5Gcreate2(m_file, group_name.c_str(), H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    H5Gclose(group);
+}
+
+void HDF5Writer::configure_attribute(const std::string& group_name, const std::string& attribute_name) {
+    std::cout << "Configuring attribute: " << attribute_name << std::endl;
```

Comment:
should we have some logging instead ?

Reply:
Yes, adding logging with spdlog is in the TODO list